### PR TITLE
add elastic tensor rotations and thermodynamic parameters

### PR DESCRIPTION
**Pull request by jan-janssen**
Move pull request https://github.com/pyiron/pyiron/pull/1203

### CODE REVIEW

File: pyiron_atomistics/atomistics/master/elastic.py [hunk 1]
```diff
@@ -28,6 +28,51 @@
...
 )
 
+def get_elastic_tensor_by_orientation(orientation, elastic_tensor):
+    """
+    Get elastic tensor in a given orientation
+    Args:
+        orientation (numpy.ndarray): 3x3 orthogonal orientation (no need to be orthonormal)
+        elastic_tensor (numpy.ndarray): 6x6 elastic tensor
+    Returns:
+        elastic_tensor in a given orientation
```
Comment (pmrv) [inline pyiron_atomistics/atomistics/master/elastic.py]: Should document the return type & dimension.  Should be clear in this case, but it's nice to be consistent.

**Comment by jan-janssen** [comment]
@samwaseda What about this pull request? I mainly just copied it from the previous pyiron repository.

**Comment by samwaseda** [comment]
Sorry I'm gonna finish the changes this week.

**Comment by samwaseda** [comment]
I guess I should be doing more about this class (especially concerning the elastic constants), but since I'm not the expert in this area, I leave it like this for now. If there's someone who wants to work on it, they can extend the class to make orientation-dependent functionality more flexible.

**Review by samwaseda — APPROVED** [review]
Looks amazing

File: pyiron_atomistics/atomistics/master/elastic.py [hunk 2]
```diff
@@ -28,6 +29,51 @@
...
+    output['elastic_tensor'] = elastic_tensor
+    output['lame_coefficient'] = np.mean(elastic_tensor[:3, :3].diagonal())
+    output['shear_modulus'] = np.mean(elastic_tensor[3:, 3:].diagonal())
+    output['bulk_modulus'] = np.mean(elastic_tensor[:3,:3])
+    output['youngs_modulus'] = 1/np.mean(np.linalg.inv(elastic_tensor[:3,:3]).diagonal())
+    output['poissons_ratio'] = -output['youngs_modulus']*np.sum(
+        np.linalg.inv(elastic_tensor[:3,:3])
+    )/6+0.5
+    output['zener_ratio'] = 12*np.mean(elastic_tensor[3:,3:].diagonal())
+    output['zener_ratio'] /= (3*np.trace(elastic_tensor[:3,:3])-np.sum(elastic_tensor[:3,:3]))
```
Comment (jan-janssen) [inline pyiron_atomistics/atomistics/master/elastic.py:74]:
```suggestion
    output = {
        'elastic_tensor': elastic_tensor,
        'lame_coefficient': np.mean(elastic_tensor[:3, :3].diagonal()),
        'shear_modulus': np.mean(elastic_tensor[3:, 3:].diagonal()),
        'bulk_modulus': np.mean(elastic_tensor[:3,:3]),
        'youngs_modulus': 1/np.mean(np.linalg.inv(elastic_tensor[:3,:3]).diagonal()),
        'zener_ratio': 12*np.mean(elastic_tensor[3:,3:].diagonal())
    }
    output['poissons_ratio'] = -output['youngs_modulus']*np.sum(
        np.linalg.inv(elastic_tensor[:3,:3])
    )/6+0.5
    output['zener_ratio'] /= (3*np.trace(elastic_tensor[:3,:3])-np.sum(elastic_tensor[:3,:3]))
```

Comment (samwaseda) [reply pyiron_atomistics/atomistics/master/elastic.py:74]: Is there any particular reason that this change should be made?

Comment (liamhuber) [reply pyiron_atomistics/atomistics/master/elastic.py:74]: If we're going to change it, could we please use `InputList`? Then dot-syntax is automatically supported, e.g. `output.shear_modulus`.

Comment (jan-janssen) [reply pyiron_atomistics/atomistics/master/elastic.py:74]: @liamhuber  I would prefer to merge this now and do the migration to input list in a separate pull request. 
@samwaseda It is just a suggestion - we do not need to change it.

**Comment by jan-janssen** [comment]
@liamhuber and @samwaseda if both of you agree I would merge this now

**Comment by samwaseda** [comment]
I was faster! 😎

**Pull request merged**